test_utils/test_utils.go

```diff
@@ -0,0 +1,375 @@
+package test_utils
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/activations"
+	"github.com/onflow/cadence/common"
+	"github.com/onflow/cadence/interpreter"
+	"github.com/onflow/cadence/pretty"
+	"github.com/onflow/cadence/sema"
+	"github.com/onflow/cadence/stdlib"
+	"github.com/onflow/cadence/test_utils/sema_utils"
+
+	"github.com/onflow/cadence/bbq/vm"
+	compilerUtils "github.com/onflow/cadence/bbq/vm/test"
+)
+
+type ParseCheckAndInterpretOptions struct {
+	Config             *interpreter.Config
+	CheckerConfig      *sema.Config
+	HandleCheckerError func(error)
+}
+
+var compile = false
+
+func init() {
+	// TODO: Set the env-variable and re-run the tests.
+	// compile = os.Getenv("COMPILE") == "true"
+}
+
+type Invokable interface {
+	interpreter.ValueComparisonContext
+	Invoke(functionName string, arguments ...interpreter.Value) (value interpreter.Value, err error)
+}
+
+type VMInvokable struct {
+	*interpreter.Interpreter
+	vmInstance *vm.VM
+	config     *vm.Config
+}
+
+var _ Invokable = &VMInvokable{}
+
+func (v *VMInvokable) Invoke(functionName string, arguments ...interpreter.Value) (value interpreter.Value, err error) {
+	vmArguments := make([]vm.Value, 0, len(arguments))
+	for _, argument := range arguments {
+		vmArguments = append(
+			vmArguments,
+			vm.InterpreterValueToVMValue(v.config.Storage, argument),
+		)
+	}
+
+	result, err := v.vmInstance.Invoke(functionName, vmArguments...)
+
+	interpreterValue := vm.VMValueToInterpreterValue(v.config, result)
+	return interpreterValue, err
+}
+
+func ParseCheckAndPrepare(t testing.TB, code string) Invokable {
+	t.Helper()
+
+	if !compile {
+		return parseCheckAndInterpret(t, code)
+	}
+
+	vmConfig := &vm.Config{}
+	vmInstance := compilerUtils.CompileAndPrepareToInvoke(
+		t,
+		code,
+		compilerUtils.CompilerAndVMOptions{
+			VMConfig: vmConfig,
+		},
+	)
+
+	return &VMInvokable{
+		vmInstance:  vmInstance,
+		config:      vmConfig,
+		Interpreter: vmConfig.Interpreter(),
+	}
+
+}
+
+// Below helper functions were copied as-is from `misc_test.go`.
+// Idea is to eventually use the below functions everywhere, and remove them from `misc_test.go`,
+// so that the `misc_test.go` would contain only tests.
```

Comment:
Similar here, I'll open a follow-up PR to refactor all existing tests to use the API in this file (still using the interpreter, not the compiler), and remove them from `misc_test.go`.

I didn't do it here because it will make this PR huge, and also is not related to "using the compiler in existing tests" (i.e: this is an unrelated refactor)